pkg/storage/db.go

```diff
@@ -701,22 +705,32 @@ func (db *DB) insertNetworkChangeEvent(ctx context.Context, timestamp time.Time,
 
 // FetchAll gets all the assets present at the specified time
 func (db *DB) FetchAll(ctx context.Context, when time.Time, count uint, offset uint, typeFilter string) ([]domain.CloudAssetDetails, error) {
-	return db.runQuery(ctx, bulkResourcesQuery, when, typeFilter, count, offset)
+	return db.runQuery(ctx, MinimumSchemaVersion, bulkResourcesQuery, when, typeFilter, count, offset)
 }
 
 // FetchByHostname gets the assets who have hostname at the specified time
 func (db *DB) FetchByHostname(ctx context.Context, when time.Time, hostname string) ([]domain.CloudAssetDetails, error) {
-	sqlstmt := fmt.Sprintf(latestStatusQuery, `aws_hostnames_hostname`)
-	return db.runQuery(ctx, sqlstmt, hostname, when)
+	ver, err := db.GetSchemaVersion(ctx)
+	if err != nil {
+		return nil, err
+	}
+	var assets []domain.CloudAssetDetails
+	if ver < DualWritesSchemaVersion {
```

Comment:
We have a separate version for reads from new schema.

Reply:
Good catch, will update that